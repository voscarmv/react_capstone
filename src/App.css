.width-auto {
  width: 800px !important;
}
