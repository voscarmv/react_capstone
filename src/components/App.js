import React from 'react';
import List from './List';
import Form from './Form';
import BreedMenu from './BreedMenu';
import BreedFilter from './BreedFilter';

const App = () => (
  <div>
    <List />
    <Form />
    <BreedFilter />
    <BreedMenu />
  </div>
);

export default App;
