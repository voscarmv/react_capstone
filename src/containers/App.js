import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Nav from './Nav';
import Cat from './Cat';
import Home from './Home';

const App = () => (
  <Router>
    <Nav />
    <Switch>
      <Route exact path="/">
        <Home />
      </Route>
      <Route path="/cat">
        <Cat />
      </Route>
    </Switch>
  </Router>
);

export default App;
