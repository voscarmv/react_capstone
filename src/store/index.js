import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import combineReducers from '../reducers/index';

const store = createStore(
  combineReducers,
  composeWithDevTools(
    applyMiddleware(
      logger,
      thunk,
    ),
  ),
);

export default store;
