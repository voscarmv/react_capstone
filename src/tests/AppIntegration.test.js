import React from 'react';
import TestRenderer, { act } from 'react-test-renderer';
// import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../store/index';
import App from '../containers/App';
import 'regenerator-runtime/runtime';

describe('App', () => {
  test('renders App component', async () => {
    let component;
    await act(async () => {
      component = TestRenderer.create(<Provider store={store}><App /></Provider>);
    });
    // screen.debug();
    // fails
    // expect(screen.getByText(/Searches for JavaScript/)).toBeNull();
    // expect(component.toJSON()).toMatchSnapshot();
  });
});
