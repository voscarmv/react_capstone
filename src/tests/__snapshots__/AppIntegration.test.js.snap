// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <span
      className="navbar-brand"
    >
      <span
        aria-label="cat"
        role="img"
      >
        😺
      </span>
      <a
        href="/"
        onClick={[Function]}
      >
        The Cat-alog
      </a>
    </span>
    <div
      className="container-fluid"
    >
      <div
        className="ml-auto text-light"
      >
        Show breed:
      </div>
      <div
        className="p-3"
      >
        <form
          className="form-inline"
        >
          <select
            className="form-control"
            onChange={[Function]}
          />
        </form>
      </div>
      <div
        className=""
      >
        <button
          aria-controls="collapse-filter-form"
          aria-expanded={false}
          className="btn btn-primary"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Filter
        </button>
      </div>
    </div>
  </nav>,
  <div
    aria-expanded={null}
    className="collapse"
    id="collapse-filter-form"
  >
    <div
      className="bg-dark text-light p-3"
    >
      <form>
        <div
          className="ml-auto mr-3 font-weight-bold"
        >
          Filter by
        </div>
        <label
          className="ml-auto"
          htmlFor="experimental"
        >
          Experimental
          <input
            className="mr-3 ml-1"
            id="experimental"
            onChange={[Function]}
            type="checkbox"
            value="experimental"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="hairless"
        >
          Hairless
          <input
            className="mr-3 ml-1"
            id="hairless"
            onChange={[Function]}
            type="checkbox"
            value="hairless"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="natural"
        >
          Natural
          <input
            className="mr-3 ml-1"
            id="natural"
            onChange={[Function]}
            type="checkbox"
            value="natural"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="rare"
        >
          Rare
          <input
            className="mr-3 ml-1"
            id="rare"
            onChange={[Function]}
            type="checkbox"
            value="rare"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="suppressed_tail"
        >
          Suppressed tail
          <input
            className="mr-3 ml-1"
            id="suppressed_tail"
            onChange={[Function]}
            type="checkbox"
            value="suppressed_tail"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="short_legs"
        >
          Short legs
          <input
            className="mr-3 ml-1"
            id="short_legs"
            onChange={[Function]}
            type="checkbox"
            value="short_legs"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="hypoallergenic"
        >
          Hypoallergenic
          <input
            className="mr-3 ml-1"
            id="hypoallergenic"
            onChange={[Function]}
            type="checkbox"
            value="hypoallergenic"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="adaptability"
        >
          Adaptability
          <input
            className="mr-3 ml-1"
            id="adaptability"
            onChange={[Function]}
            type="checkbox"
            value="adaptability"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="affection_level"
        >
          Affection level
          <input
            className="mr-3 ml-1"
            id="affection_level"
            onChange={[Function]}
            type="checkbox"
            value="affection_level"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="child_friendly"
        >
          Child friendly
          <input
            className="mr-3 ml-1"
            id="child_friendly"
            onChange={[Function]}
            type="checkbox"
            value="child_friendly"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="dog_friendly"
        >
          Dog friendly
          <input
            className="mr-3 ml-1"
            id="dog_friendly"
            onChange={[Function]}
            type="checkbox"
            value="dog_friendly"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="energy_level"
        >
          Energy level
          <input
            className="mr-3 ml-1"
            id="energy_level"
            onChange={[Function]}
            type="checkbox"
            value="energy_level"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="grooming"
        >
          Grooming
          <input
            className="mr-3 ml-1"
            id="grooming"
            onChange={[Function]}
            type="checkbox"
            value="grooming"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="health_issues"
        >
          Health issues
          <input
            className="mr-3 ml-1"
            id="health_issues"
            onChange={[Function]}
            type="checkbox"
            value="health_issues"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="intelligence"
        >
          Intelligence
          <input
            className="mr-3 ml-1"
            id="intelligence"
            onChange={[Function]}
            type="checkbox"
            value="intelligence"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="shedding_level"
        >
          Shedding level
          <input
            className="mr-3 ml-1"
            id="shedding_level"
            onChange={[Function]}
            type="checkbox"
            value="shedding_level"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="social_needs"
        >
          Social needs
          <input
            className="mr-3 ml-1"
            id="social_needs"
            onChange={[Function]}
            type="checkbox"
            value="social_needs"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="stranger_friendly"
        >
          Stranger friendly
          <input
            className="mr-3 ml-1"
            id="stranger_friendly"
            onChange={[Function]}
            type="checkbox"
            value="stranger_friendly"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="vocalisation"
        >
          Vocalisation
          <input
            className="mr-3 ml-1"
            id="vocalisation"
            onChange={[Function]}
            type="checkbox"
            value="vocalisation"
          />
        </label>
        <hr />
        <div
          className="ml-auto mr-3 font-weight-bold"
        >
          Filter preferences
        </div>
        <label
          className="ml-auto"
          htmlFor="experimental"
        >
          <span>
            Experimental
          </span>
          <input
            className="mr-3 ml-1"
            disabled={true}
            id="experimental"
            onChange={[Function]}
            type="checkbox"
            value="experimental"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="hairless"
        >
          <span>
            Hairless
          </span>
          <input
            className="mr-3 ml-1"
            disabled={true}
            id="hairless"
            onChange={[Function]}
            type="checkbox"
            value="hairless"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="natural"
        >
          <span>
            Natural
          </span>
          <input
            className="mr-3 ml-1"
            disabled={true}
            id="natural"
            onChange={[Function]}
            type="checkbox"
            value="natural"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="rare"
        >
          <span>
            Rare
          </span>
          <input
            className="mr-3 ml-1"
            disabled={true}
            id="rare"
            onChange={[Function]}
            type="checkbox"
            value="rare"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="suppressed_tail"
        >
          <span>
            Suppressed tail
          </span>
          <input
            className="mr-3 ml-1"
            disabled={true}
            id="suppressed_tail"
            onChange={[Function]}
            type="checkbox"
            value="suppressed_tail"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="short_legs"
        >
          <span>
            Short legs
          </span>
          <input
            className="mr-3 ml-1"
            disabled={true}
            id="short_legs"
            onChange={[Function]}
            type="checkbox"
            value="short_legs"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="hypoallergenic"
        >
          <span>
            Hypoallergenic
          </span>
          <input
            className="mr-3 ml-1"
            disabled={true}
            id="hypoallergenic"
            onChange={[Function]}
            type="checkbox"
            value="hypoallergenic"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="adaptability"
        >
          <span>
            Adaptability
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="adaptability"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="affection_level"
        >
          <span>
            Affection level
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="affection_level"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="child_friendly"
        >
          <span>
            Child friendly
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="child_friendly"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="dog_friendly"
        >
          <span>
            Dog friendly
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="dog_friendly"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="energy_level"
        >
          <span>
            Energy level
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="energy_level"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="grooming"
        >
          <span>
            Grooming
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="grooming"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="health_issues"
        >
          <span>
            Health issues
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="health_issues"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="intelligence"
        >
          <span>
            Intelligence
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="intelligence"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="shedding_level"
        >
          <span>
            Shedding level
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="shedding_level"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="social_needs"
        >
          <span>
            Social needs
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="social_needs"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="stranger_friendly"
        >
          <span>
            Stranger friendly
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="stranger_friendly"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
        <label
          className="ml-auto"
          htmlFor="vocalisation"
        >
          <span>
            Vocalisation
          </span>
          <input
            className="mr-3 ml-1"
            defaultValue="1"
            disabled={true}
            id="vocalisation"
            max="5"
            min="1"
            onChange={[Function]}
            type="number"
          />
        </label>
      </form>
    </div>
  </div>,
  <div
    className="m-3"
  >
    <h5>
      Welcome to The Cat-alog!
      <span
        aria-label="cat"
        role="img"
      >
        🐈
      </span>
    </h5>
    <p>
      You can browse through our database of cat breeds by using the filter on the top-right corner of this page.
    </p>
    <p>
      Simply select a breed from the drop-down menu to display its details here. Or if you prefer, filter them first by clicking on the Filter button and setting the parameters you sre looking for!
    </p>
  </div>,
]
`;
